torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -265,9 +266,7 @@ Tensor & VariableType::s_copy_(Tensor & self, const Tensor & src, bool non_block
   }
   {
     at::AutoNonVariableTypeMode non_var_type_mode(true);
-    if (self.is_sparse() && src.is_sparse()) baseType->copy_sparse_to_sparse_(self_, src_, non_blocking);
-    else if (!self.is_sparse() && !src.is_sparse()) baseType->s_copy_(self_, src_, non_blocking);
-    else AT_ERROR("copy_() between dense and sparse Tensors is not implemented! Found self type = ", self.type(), " and src type = ", src.type());
+    baseType->copy_(self_, src_, non_blocking);
```

Comment:
I think the main reason to do this manually now is because we don't have outplace copy, so we need to special case in the tracer for the force_outplace case.

Reply:
can you add a comment so we know why this was done?